src/index.js

```diff
@@ -1,32 +1,39 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const API = 'https://rickandmortyapi.com/api/character/';
+const xhttp = new XMLHttpRequest();
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
+function fetchData(url_api) {
+  return new Promise ((resolve, reject) => {
+    xhttp.onreadystatechange = (event) => {
+      if (xhttp.readyState === 4 && xhttp.status == 200) {
+        resolve(JSON.parse(xhttp.responseText))
+      }
     }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
+    xhttp.open('GET', url_api, false);
+    xhttp.send();
+    reject(onError)
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```